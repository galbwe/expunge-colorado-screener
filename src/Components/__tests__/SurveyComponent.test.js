import * as Survey from 'survey-react';

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import SURVEY_DATA from '../../data/survey';
import SurveyComponent from '../SurveyComponent';
import jsSHA from 'jssha';

Enzyme.configure({ adapter: new Adapter() });

function createSurveyVersion(data) {
    // create survey version # by hashing stringified survey data
    const shaObj = new jsSHA('SHA-1', 'TEXT');
    shaObj.update(JSON.stringify(data));
    return shaObj.getHash('HEX');
}

test('survey renders', () => {
    const surveyModel = new Survey.Model(SURVEY_DATA);
    const component = shallow(
        <SurveyComponent
            surveyModel={surveyModel}
            version={createSurveyVersion(SURVEY_DATA)}
        />
    );
    expect(component).toMatchSnapshot();
});
