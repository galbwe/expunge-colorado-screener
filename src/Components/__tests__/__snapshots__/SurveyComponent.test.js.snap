// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`survey renders 1`] = `ShallowWrapper {}`;
